src/main/java/org/snomed/snowstorm/core/data/services/BranchReviewService.java

```diff
@@ -256,22 +256,30 @@ private Concept upgradeComponents(Concept sourceVersion, Concept targetVersion)
 
 	private <T extends SnomedComponent> Set<T> mergePublished(Set<T> sourceComponents, Set<T> targetComponents) {
 		Map<String, T> mergedComponents = new HashMap<>();
-
 		for (T sourceComponent : sourceComponents) {
-			boolean publishedAndUnchanged = sourceComponent.getReleasedEffectiveTime() != null && sourceComponent.getEffectiveTimeI() != null;
-			if (publishedAndUnchanged) {
+			if (sourceComponent.getEffectiveTimeI() != null) {
```

Comment:
@CoderMChu If I do not exclude unpublished components, then several tests fail because the task will receive the project's changes despite the author picking the RHS. 

- testUpgradeExtensionWhenExtensionProjectAndTaskBothHaveAuthoringChangesAndUserSelectsRHS
- testChangingComponentAfterPublishingAndUserSelectsRHS
- testUpgradeExtensionWhenExtensionProjectAndTaskBothHaveAuthoredSameComponentAndUserSelectsRHS

Reply:
@dmcgihtsdo  I had a look at the first test case above. The failed assertion doesn't look be correct. Let's have a quick call.